tests/e2e/workloads/couchbase/test_couchbase_node_reboot.py

```diff
@@ -49,34 +53,56 @@ def cb_setup(self, couchbase_factory_fixture):
             )
         ]
     )
-    def test_run_couchbase_node_reboot(self, cb_setup, nodes, pod_name_of_node):
+    def test_run_couchbase_node_reboot(
+            self, cb_setup, nodes, pod_name_of_node):
         """
         Test couchbase workload with node reboot
         """
+        # Check worker node utilization (adm_top)
+        get_node_resource_utilization_from_adm_top(
+            node_type='worker', print_table=True
+        )
+        get_node_resource_utilization_from_adm_top(
+            node_type='master', print_table=True
+        )
+
         if pod_name_of_node == 'couchbase':
             node_list = self.cb.get_couchbase_nodes()
         elif pod_name_of_node == 'osd':
             node_list = get_osd_running_nodes()
         elif pod_name_of_node == 'master':
-            node_list = get_master_nodes()
+            master_node = get_typed_nodes(pod_name_of_node, num_of_nodes=1)
 
-        node_1 = get_node_objs(
-            node_list[random.randint(0, len(node_list) - 1)])
+        # Restart relevant node
+        if pod_name_of_node == 'master':
+            nodes.restart_nodes(master_node, wait=False)
+            waiting_time = 40
+            log.info(f"Waiting {waiting_time} seconds...")
```

Comment:
why are we waiting here? if it's for the nodes to come up to Ready state, we have a function for that in node.py - wait_for_nodes_status

Reply:
I referred this https://github.com/red-hat-storage/ocs-ci/blob/master/tests/e2e/workloads/amq/test_amq_node_reboot_and_shutdown.py#L104
in master node reboot, we don't have checks for checking cluster connectivity in restart_nodes()
So I skipped wait in restart_nodes(), and kept wait_for_nodes_status() and wait_for_cluster_connectivity() after restart.

Reply:
you can use ocs_ci.ocs.ocp.wait_for_cluster_connectivity

Reply:
Yes , I'm using it

Reply:
but nodes.restart_nodes() has a wait functionality - ocs_ci.ocs.node.wait_for_nodes_status

What am I missing?

Reply:
Rebooting master sometimes it fails with an unauthorized error. So not waiting there and in below lines trying to catch ResourceWrongStatusException errors and untill node becomes ready state. So using  wait_for_nodes_status() after some sleep.